src/univers/gem.py

```diff
@@ -242,6 +237,33 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self.__cmp__(other) >= 0
 
+    @property
+    def segments(self):
+        """
+        Return a new sequence of segments for this version where segments are
+        ints or strings parsed from the original version string.
+        """
+        find_segments = re.compile(r"[0-9]+|[a-z]+", re.IGNORECASE).findall
+        segments = []
+        for seg in find_segments(self.version):
+            if seg.isdigit():
+                seg = int(seg)
+            segments.append(seg)
+        return segments
+
+    @property
+    def canonical_segments(self):
```

Comment:
a yield looks simpler:

```suggestion
    def canonical_segments(self):
        """
        Yield  "canonical segments" for this version using
        the Rubygems way for canonicalization.
        """
        for segments in self.split_segments():
            segs = dropwhile(lambda s: s == 0, reversed(segments))
            yield from sorted(segs, reverse=True)
```